docs/cookbooks/studio/usecases-functions.md

```diff
@@ -0,0 +1,137 @@
+
+![](https://i.imgur.com/dKN4rrr.png)
+
+4. Navigate to **Functions** and pass the variable containing the API response and write a code that matches the dynamic format of the carousel node. Click [here](https://docs.yellow.ai/docs/platform_concepts/studio/build/nodes/message-nodes#4-carousel) for the dynamic format.
+
+![](https://i.imgur.com/Jklf0Ia.png)
+
+5. Include a [Function node](https://docs.yellow.ai/docs/platform_concepts/studio/build/nodes/action-nodes#24-function) and choose the function you created in the previous step. Save the response of this node into another variable.
+
+![](https://i.imgur.com/PQhxMKE.png)
+
+6. Add a [carousel node](https://docs.yellow.ai/docs/platform_concepts/studio/build/nodes/message-nodes#4-carousel) to the flow. In the **Fetch from** field of the carousel node, fetch the variable created in the previous step to populate the carousel with the desired data.
+
+![](https://i.imgur.com/Mlh4Xzv.png)
+
+:::note
+Instead of including the Function node, you can also pass the function in the **Parse API response** field.
+![](https://i.imgur.com/eIReCNg.png)
+:::
+
+
+## Calculate 45 days before the present date
+
+To achieve this, we will retrieve the timestamp from the **Date** variable in the platform.
+
+```
+{
+    //Date object
+    
+    "timestamp": "Wed, 28 Jul 2022 05:04:18 GMT",
+    "year": 2022,
+    "month": 7,
+    "date": 28,
+    "day": "Thursday",
+    "hour": 5,
+    "minute": 4
+}
+```
+
+1. [Create a flow](https://docs.yellow.ai/docs/platform_concepts/studio/build/Flows/journeys#2-create-a-flow) and add a [variable node](https://docs.yellow.ai/docs/platform_concepts/studio/build/nodes/action-nodes#22-variables). In the node, create a variable, **bookindate** and set the value as {{{date.timestamp}}}.
+
+![](https://i.imgur.com/e2JnHnf.png)
+
+2. Add a [text node](https://docs.yellow.ai/docs/platform_concepts/studio/build/nodes/message-nodes#2-text) and pass the variable **bookindate** in it. 
+
+![](https://i.imgur.com/aQMaaUM.png)
+
+3. Write the following code and pass the function into a node. Store it's response in a variable.
+
+```
+return new Promise(resolve => {
+    let bookingDate = data.variables.bookingDate;
+    let dt = new Date(bookingDate);
+
+    dt.setDate(dt.getDate() - 45);
+    console.log("Final Timestamp ---> ", dt.toLocaleString());
+
+    let pdate = dt.toLocaleString()
+    console.log(pdate,"PDATE IS --->");
+    const myArraydate = pdate.split(",");
+    const myArray = myArraydate[0].split("/");
+    let newDate = myArray[1] + "/" + myArray[0] + "/" + myArray[2]
+    data.variables.bookingDate = newDate;
+
+    resolve(newDate);
+});    
+```
+![](https://i.imgur.com/GIQH8Op.png)
+
+
+4. Pass that variable in the text node to display the result to the end user.
+
+![](https://i.imgur.com/2cCX1By.png)
+
+**Result:**
+
+<img src="https://i.imgur.com/fioBFmp.png" alt="drawing" width="60%"/>
+
+
+## Codes to perform sentence autocompletion and decode BASE64
```

Comment:
add this as separate headings. 

## Function (code) to enable autocompletion
## Function (code) to decode BASE64

Reply:
separated the headings